R/simulate_design.R

```diff
@@ -128,17 +131,12 @@ simulate_single_design <- function(design, sims) {
   } else {
     sims <- check_sims(design, sims)
     results_list <- fan_out(design, sims)
-    # fan_id <- setNames(
-    #   lapply(rev(sims$n), seq),
-    #   paste0("fan_", rev(seq_len(nrow(sims))))
-    # )
-    # fan_id <- expand.grid(rev(fan_id))
-    s <- setNames(sims$n, paste0("step_", sims$end, "_draw"))
-    fan_id <- do.call(cbind.data.frame, lapply(
+    sims2    <- rbind(sims[1,], sims[-1,][sims[-1,]$n!=1,]) # preserve first row; drop all others with n = 1
```

Comment:
this seems like a workaround for a underlying issue in check_sims ? Should it get fixed there instead.

Reply:
Yes agree.

Reply:
My preference is to get this bug fixed asap and then improve the code.